### ISSUE TITLE
GHC 7.10 support

### ISSUE THREAD

**Issue by taktoa** [cjdev/monad-mock#1]
I am using this library in `language-ninja`, which attempts to support all versions of GHC going back to the version used in the latest Ubuntu LTS release (right now, that's GHC 7.10.3). Unfortunately, `monad-mock` currently contains some Template Haskell stuff that seems difficult to make functional across many versions of GHC. It seems to me that it would be a good idea to have a separate `monad-mock-th`, to avoid this issue.

If you'd rather just fix `monad-mock` for 7.10, here's a diff representing the progress I made on that front:

```
diff --git a/.stylish-haskell.yaml b/.stylish-haskell.yaml
new file mode 100644
index 0000000..5e36f0c
--- /dev/null
+++ b/.stylish-haskell.yaml
@@ -0,0 +1 @@
+steps: {}
diff --git a/library/Control/Monad/Mock/TH.hs b/library/Control/Monad/Mock/TH.hs
index 20fc9e6..8180528 100644
--- a/library/Control/Monad/Mock/TH.hs
+++ b/library/Control/Monad/Mock/TH.hs
@@ -1,5 +1,9 @@
 {-# LANGUAGE CPP #-}
+#if __GLASGOW_HASKELL__ >= 801
 {-# LANGUAGE TemplateHaskellQuotes #-}
+#else
+{-# LANGUAGE TemplateHaskell #-}
+#endif
 
 {-|
 This module provides Template Haskell functions for automatically generating
diff --git a/package.yaml b/package.yaml
index 2ea3c77..663cb49 100644
--- a/package.yaml
+++ b/package.yaml
@@ -40,7 +40,7 @@ default-extensions:
 
 library:
   dependencies:
-  - base >= 4.9.0.0 && < 5
+  - base >= 4.6.0.0 && < 5
   - constraints >= 0.3.1
   - exceptions >= 0.6
   - haskell-src-exts
@@ -48,7 +48,8 @@ library:
   - th-orphans
   - monad-control >= 1.0.0.0 && < 2
   - mtl
-  - template-haskell >= 2.11.0.0 && < 2.13
+  - template-haskell >= 2.10.0.0 && < 2.13
+  - transformers
   - transformers-base
   source-dirs: library
 ```

It currently fails due to changes in the `Infix*` and `Parens*` constructors between `template-haskell-2.10.0.0` and `template-haskell-2.11.0.0`.

**Comment by lexi-lambda** [comment]
I think I’d much rather backport `monad-mock` to work with older versions rather than split its already-small interface into unnecessarily small fragments. There are significant advantages to having a cohesive interface—the npm ecosystem has plenty of examples where too much fragmentation became a problem.

I’ll look into backporting this library to GHC 7.10 when I get the chance.

**Comment by lexi-lambda** [comment]
Fixed in the 0.1.1.2 release.
